lib/bolt/inventory/target.rb

```diff
@@ -109,14 +109,19 @@ def invalidate_config_cache!
       def transport_config_cache
         if @transport_config_cache.nil?
           merged_config = Bolt::Util.deep_merge(group_cache['config'], @config)
-          # Use the base config to ensure we handle the config validation and
-          # munging correctly
-          config = @inventory.config.deep_clone
-          config.update_from_inventory(merged_config)
-          config.validate
+          base_config = @inventory.config
+          transport_data = Bolt::Util.deep_clone(base_config.transports)
+          Bolt::Config.update_transport_hash(base_config.boltdir.path, transport_data, merged_config)
```

Comment:
This change is a bit weird, but it will be superseded by work that @beechtom is doing around how we handle transport config, so it seems worth the performance improvement for now.